src/main/java/StudentGroup.java

```diff
@@ -0,0 +1,57 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class StudentGroup {
+
+    private Student groupLeader;
+    private List<Student> students = new ArrayList<>();
+    private final List<String> tasks = new ArrayList<>();
+    Map<String, List<Student>> completedTasks = new HashMap<>();
+
+
+    public StudentGroup(Student groupLeader) {
+        this.groupLeader = groupLeader;
+    }
+
+    public Student changeGroupLeader(Student newGroupLeader) {
+        this.groupLeader = newGroupLeader;
+        return newGroupLeader;
+    }
+
+    public void addStudent(Student student) {
+        this.students.add(student);
```

Comment:
Before adding student to list it would be great to check that it is not there yet